File: src/plugins/shade.ts
import { IMatrixClient, Plugin, CommandDef, MessageContext } from "./base";

const COMING_SOON = "This feature is coming soon.";

export class ShadePlugin extends Plugin {
  constructor(client: IMatrixClient) {
    super(client);
  }

  get name() {
    return "shade";
  }

  get commands(): CommandDef[] {
    return [
      { name: "shadecheck", description: "(stub) Coming soon", usage: "!shadecheck [@user]" },
      { name: "shadeboard", description: "(stub) Coming soon" },
      { name: "perpetrators", description: "(stub) Coming soon" },
      { name: "receipts", description: "(stub) Coming soon" },
      { name: "shadewar", description: "(stub) Coming soon" },
    ];
  }

  async onMessage(ctx: MessageContext): Promise<void> {
    const stubCommands = ["shadecheck", "shadeboard", "perpetrators", "receipts", "shadewar"];
    for (const cmd of stubCommands) {
      if (this.isCommand(ctx.body, cmd)) {
        await this.sendReply(ctx.roomId, ctx.eventId, COMING_SOON);
        return;
      }
    }
  }
}

File: internal/plugin/shade.go
package plugin

import (
	"os"

	"maunium.net/go/mautrix"
)

// ShadePlugin is a stub plugin for future shade/roast features.
type ShadePlugin struct {
	Base
	enabled bool
}

// NewShadePlugin creates a new shade plugin.
func NewShadePlugin(client *mautrix.Client) *ShadePlugin {
	return &ShadePlugin{
		Base:    NewBase(client),
		enabled: os.Getenv("FEATURE_SHADE") == "true" || os.Getenv("FEATURE_SHADE") == "1",
	}
}

func (p *ShadePlugin) Name() string { return "shade" }

func (p *ShadePlugin) Commands() []CommandDef {
	if !p.enabled {
		return nil
	}
	return []CommandDef{
		{Name: "shade", Description: "Throw some shade (coming soon)", Usage: "!shade", Category: "LLM & Sentiment"},
	}
}

func (p *ShadePlugin) Init() error { return nil }

func (p *ShadePlugin) OnReaction(_ ReactionContext) error { return nil }

func (p *ShadePlugin) OnMessage(ctx MessageContext) error {
	if !p.enabled {
		return nil
	}

	if p.IsCommand(ctx.Body, "shade") {
		return p.SendReply(ctx.RoomID, ctx.EventID, "Coming soon.")
	}

	return nil
}
